src/live-programming.rst

```diff
@@ -146,35 +121,33 @@ Some research dig more about the programmable bricks and make more developments.
   .. bibliography:: programmable_PE2.bib
     :filter: key == "gindling1995lego"
 
+  The paper presents an educational environment, LEGOsheets. It works with the prgrammable brick to provide a gentle introduction to programming and the design of mechanical artifacts.
+
 Industrial Automation
 ~~~~~~~~~~~~~~~~~~~~~
 
 The automation of industrial manufacture has a long history of development. In the 20th century, industrial automation was well advanced in two areas of manufacture, including homogeneous products and mass manufacture of discrete products. In old time, the industrial automation was still a kind of "hard" or "fixed" automation, which doesn't enable much flexibility.
 
-.. container:: bib-item
-
-  .. bibliography:: programmable_PE2.bib
-    :filter: key == "saraswat1994factory"
-
-  The article describes a rapid thermal multiprocessing for a programmable factory.
-
 .. container:: bib-item
 
   .. bibliography:: programmable_PE3.bib
     :filter: key == "nitzan1976industrial"
 
   The article gives an early idea of industrial automation. It also gives a classification on automation applied in industry.
 
-As the development of technology, industrial automation enables more flexible work, which enables the same automated equipment to be programmed to perform different tasks. The development of the controller systems gave the growth of research on designing corresponding interface for the controller systems. Here are some papers on different interface architecture for programmable controller systems.
+.. container:: bib-item
+
+  .. bibliography:: programmable_PE2.bib
+    :filter: key == "saraswat1994factory"
 
-For now, there are many researches on programmable logic circuit. For college learning and industrial production, software for programming logic circuits is widely used.
+  The paper presents a research on the development of concepts of a programmable factory and an altemative adaptable manufacturing systems approach to integrated circuit fabrication. The approach can offer more economical small or large scale production and higher flexibility to accommodate many products on several processes.
 
-However, it is quite hard to divide these application as a branch of programmable physical environment. Although industrial robots are programmable, doing so remains difficult. Making environments shared between humans and robots more programmable remains an area of ongoing and future research.
+However, many physical machines are not or hard to be programmable in an industrial environment. Many of the machines still need traditional pre-programming methods to ensure stable functionalities. Making environments shared between humans and robots more programmable remains an area of ongoing and future research.
```

Comment:
"many physical machines are not or hard to be programmable in an industrial environment" huh?